cli-plugins/apps/how-to-guides.hbs.md

```diff
@@ -0,0 +1,611 @@
+# <a id='how-to-guides'>How-to-guides
+
+<!-- Intro to this point -->
+
+<!-- ## <a id='service-account-usage'> --service-account usage -->
+
+## <a id='custom-registry'> Custom registry credentials
+
+Apps CLI plug-in allows users to push images to their private registry by setting some flags when creating from local source code. 
+User can either trust a custom certificate in their system or pass the path to the certificate through flags.
+
+To apply a workload with this method, user may need to specify:
+
+- `--registry-ca-cert`, which refers to the path of the self-signed certificate needed for the custom/private registry. This is also populated with a default value through the environment variable `TANZU_APPS_REGISTRY_CA_CERT`.
+- `--registry-password` which is used when the registry requires credentials to push. The value of this flag can also be specified through `TANZU_APPS_REGISTRY_PASSWORD`.
+- `--registry-username` usually used with `--registry-password` to set the registry credentials. It can also be provided as the environment variable `TANZU_APPS_REGISTRY_USERNAME`. 
+- `--registry-token` which is set when the registry authentication is done via token. The value of this flag can also be taken from `TANZU_APPS_REGISTRY_TOKEN` environment variable.
+
+For example:
+
+```bash
+tanzu apps workload apply my-workload --local-path path/to/my/repo -s registry.url.nip.io/my-package/my-image --type web --registry-ca-cert path/to/cacert/mycert.nip.io.crt --registry-username my-username --registry-password my-password
+❓ Publish source in "path/to/my/repo" to "registry.url.nip.io/my-package/my-image"? It may be visible to others who can pull images from that repository [Yn]: y
+Publishing source in "path/to/my/repo" to "registry.url.nip.io/my-package/my-image"...
+37.53 kB / 37.53 kB [-----------------------------------------------------------------------------------] 100.00% 57.67 kB p/s
+📥 Published source
+
+🔎 Create workload:
+      1 + |---
+      2 + |apiVersion: carto.run/v1alpha1
+      3 + |kind: Workload
+      4 + |metadata:
+      5 + |  labels:
+      6 + |    apps.tanzu.vmware.com/workload-type: web
+      7 + |  name: my-workload
+      8 + |  namespace: default
+      9 + |spec:
+     10 + |  source:
+     11 + |    image: registry.url.nip.io/my-package/my-image:latest@sha256:caeb7e3a0e3ae0659f74d01095b6fdfe0d3c4a12856a15ac67ad6cd3b9e43648
+❓ Do you want to create this workload? [yN]:
+```
+
+Also, the same command could be run as:
+
+```bash
+export TANZU_APPS_REGISTRY_CA_CERT=path/to/cacert/mycert.nip.io.crt
+export TANZU_APPS_REGISTRY_PASSWORD=my-username
+export TANZU_APPS_REGISTRY_USERNAME=my-password
+
+tanzu apps workload apply my-workload --local-path path/to/my/repo -s registry.url.nip.io/my-package/my-image
+```
+
+This way, if there are various images that need to be pushed to the same registry, the commands to create the workloads won't need to specify the same values each time.
```

Comment:
Using environment variables provides the added convenience of not having to enter these flag values repeatedly in the event that multiple workloads must be created, with references to the same registry, during a given terminal session.